fix mobile padding on unified design picker

roo2 opened the pull request:
#### Proposed Changes
Tiny PR to fix the row gap on the mobile version of the unified design picker. Fixes https://github.com/Automattic/wp-calypso/issues/66160

Before: 
<img width="239" alt="Screen Shot 2022-08-03 at 3 31 38 pm" src="https://user-images.githubusercontent.com/22446385/182532925-02c72c70-9ccd-466b-bf30-3caaa32337bc.png">

After:
<img width="239" alt="Screen Shot 2022-08-03 at 3 31 32 pm" src="https://user-images.githubusercontent.com/22446385/182532947-8418fc8d-44a4-4e13-9244-3ea1ff32ecfd.png">

Desktop is unaffected, as it had 40px row gap already.
#### Testing Instructions
Load design picker in mobile & desktop

roo2 committed:
fix mobile padding

taipeicoder commented on packages/design-picker/src/components/style.scss:
```diff
@@ -388,6 +388,7 @@
 	}
 	.design-picker__grid {
 		@supports ( display: grid ) {
+			row-gap: 40px;
```
Currently, the gap between generated designs seems to be `80px`, can we make it `80px` here as well?
![Screen Shot 2022-08-03 at 3 44 06 PM](https://user-images.githubusercontent.com/797888/182552828-f271864a-f1a0-47a7-93d2-8caa0aa60fc5.png)

bangank36 replied:
When checking your note I realized the blank space on the left of category menu, should we make them left-aligned too?
<img width="425" alt="image" src="https://user-images.githubusercontent.com/10071857/182574016-e7c5db11-246e-4871-83ac-be090ee8cdc6.png">

taipeicoder replied:
The design shows the same padding, so it should be fine 👍
![Screen Shot 2022-08-03 at 6 01 36 PM](https://user-images.githubusercontent.com/797888/182581694-68b188df-ebe8-478e-b644-c0747579f697.png)

roo2 replied:
> Currently, the gap between generated designs seems to be 80px, can we make it 80px here as well

@taipeicoder, wait where do you get 80px from? when I look at what's currently in https://horizon.wordpress.com/ it seems to be only 24px?

arthur791004 replied:
> When checking your note I realized the blank space on the left of category menu, should we make them left-aligned too?

Yes, I think we have to remove the `padding-left` on the mobile to align the figma

![image](https://user-images.githubusercontent.com/13596067/182755903-b6f8684b-b424-4968-93ed-e73d3239484a.png)

bangank36 replied:
Okay, it should not be blocker for this, I will create seperate issue and update trigage sheet

bangank36 replied:
> > Currently, the gap between generated designs seems to be 80px, can we make it 80px here as well
> 
> @taipeicoder, wait where do you get 80px from? when I look at what's currently in https://horizon.wordpress.com/ it seems to be only 24px?

It's weird, I can also get the value of row-gap 80px yesterday, but now it is 24px again 🤔. Tested on trunk and horizon as well

arthur791004 replied:
> Currently, the gap between generated designs seems to be 80px, can we make it 80px here as well

Yes, I cannot find it either 🤔

BTW, I think we have to calculate the gap between the premium badge and it's `36px` on the Figma.

![image](https://user-images.githubusercontent.com/13596067/182757002-0dbb6c65-8b2c-4d00-847a-30c0f518c3e0.png)

Also, we have to check with another team (I forgot which team...) why the `design-picker__pricing-description` element uses absolute positioning and it will cause the gap looks weird.

roo2 replied:
@arthur791004 It looks like they were added here https://github.com/Automattic/wp-calypso/pull/65052 without discussion of absolute vs static position, I imagine it was just what worked at the time. Ok, hmm 36 px to the badge would make it 32px row gap, I think that looks a bit too small:
<img width="573" alt="Screen Shot 2022-08-04 at 2 53 31 pm" src="https://user-images.githubusercontent.com/22446385/182766065-d8bb3713-6d4b-42a9-94c6-3f3ef3b575dc.png">

taipeicoder replied:
> @taipeicoder, wait where do you get 80px from? when I look at what's currently in https://horizon.wordpress.com/ it seems to be only 24px? 

Just checked again and it's indeed 24px 🤔. Never mind!

arthur791004 replied:
>  Ok, hmm 36 px to the badge would make it 32px row gap, I think that looks a bit too small:

Yes but I think if we use a static position for the pricing description, it would be ok. What do you think? Or maybe we have to ask @saygunnyc 's preference

| | |
| - | - |
| ![image](https://user-images.githubusercontent.com/13596067/182774368-1bf527cf-2e87-4696-acb0-6c058d1410d0.png) | ![image](https://user-images.githubusercontent.com/13596067/182774467-eb6b1dfc-8f20-4fe5-8bf4-3d696d2c363c.png) |

roo2 replied:
hmm yea, I think that making the pricing description static positioned is the best solution rather than adding extra padding to compensate for it being absolute. @valterlorran, @markbiek, does that make sense to you two?

roo2 committed:
make pricing description labels static

roo2 commented:
I've updated the PR to make the pricing description static and adjusted row gaps.

Making the description static added 20px so I subtracted 20px from the desktop row gap.

This affects the existing static design picker as well as the new unified one: 

Static design picker:

<img width="181" alt="Screen Shot 2022-08-04 at 4 51 33 pm" src="https://user-images.githubusercontent.com/22446385/182784368-8f4591c4-d7be-4a04-a3eb-94321b3e2779.png">
<img width="842" alt="Screen Shot 2022-08-04 at 4 51 19 pm" src="https://user-images.githubusercontent.com/22446385/182784431-9eff66bf-5092-4e9e-9c7e-100cdebbd35a.png">

Unified DP:
<img width="309" alt="Screen Shot 2022-08-04 at 5 04 30 pm" src="https://user-images.githubusercontent.com/22446385/182784703-0b83cb07-c369-4308-9d98-4615a6725c91.png">

<img width="505" alt="Screen Shot 2022-08-04 at 5 04 01 pm" src="https://user-images.githubusercontent.com/22446385/182784602-c9179402-a0fa-4f0b-89af-5bc502b358ff.png">

arthur791004 commented on packages/design-picker/src/components/style.scss:
```diff
@@ -231,8 +231,6 @@
 	.design-picker__pricing-description {
 		text-align: left;
 		color: #50575e;
-		z-index: 1;
```
tested and the link is still clickable!

roo2 replied:
good point I actually hadn't thought to test that 😅

arthur791004 reviewed (approved):
LGTM, let's wait for the @Automattic/serenity team's review!

valterlorran reviewed (approved):
LGTM!

autumnfjeld commented:
Hm, I'm not sure there is enough padding added to make it distinct enough that the text belongs to the theme above it.  I would add more padding. 🤔

![image](https://user-images.githubusercontent.com/5665959/183911957-d73857b4-9ca4-4d90-bdde-3618e7010fac.png)

saygunnyc commented:
> Hm, I'm not sure there is enough padding added to make it distinct enough that the text belongs to the theme above it. I would add more padding. 🤔
> 
> ![image](https://user-images.githubusercontent.com/5665959/183911957-d73857b4-9ca4-4d90-bdde-3618e7010fac.png)

💯 Can we try 32px?
